src/promptflow-devkit/promptflow/_sdk/_utilities/chat_utils.py

```diff
@@ -26,6 +29,33 @@ def construct_flow_absolute_path(flow: str) -> str:
     return (flow_dir / flow_file).absolute().resolve().as_posix()
 
 
+@contextlib.contextmanager
+def generate_yaml_entry_under_flow_directory(entry: Union[str, PathLike, Callable], code: Path = None):
```

Comment:
May I ask what's the difference of generate_yaml_entry and generate_yaml_entry_under_flow_directory. Maybe can use generate_yaml_entry  directly?

Reply:
`generate_yaml_entry` will generate yaml in a temp folder, while we need to generate it under flow directory in current scenario; on the other hand, we should avoid doing this if possible, so I haven't added a parameter to the origin function.

Reply:
`generate_yaml_entry` will generate yaml in cwd(I think it's flow dir in this scenerio) if code is None

Reply:
update the implementation